config/lang/builtins.go

```diff
@@ -15,11 +15,109 @@ func registerBuiltins(scope *ast.BasicScope) *ast.BasicScope {
 	if scope.FuncMap == nil {
 		scope.FuncMap = make(map[string]ast.Function)
 	}
+
+	// Implicit conversions
+	scope.FuncMap["__builtin_FloatToInt"] = builtinFloatToInt()
+	scope.FuncMap["__builtin_FloatToString"] = builtinFloatToString()
+	scope.FuncMap["__builtin_IntToFloat"] = builtinIntToFloat()
 	scope.FuncMap["__builtin_IntToString"] = builtinIntToString()
 	scope.FuncMap["__builtin_StringToInt"] = builtinStringToInt()
+
+	// Math operations
+	scope.FuncMap["__builtin_IntMath"] = builtinIntMath()
+	scope.FuncMap["__builtin_FloatMath"] = builtinFloatMath()
 	return scope
 }
 
+func builtinFloatMath() ast.Function {
+	return ast.Function{
+		ArgTypes:     []ast.Type{ast.TypeInt},
+		Variadic:     true,
+		VariadicType: ast.TypeFloat,
+		ReturnType:   ast.TypeFloat,
+		Callback: func(args []interface{}) (interface{}, error) {
+			op := args[0].(ast.ArithmeticOp)
+			result := args[1].(float64)
+			for _, raw := range args[2:] {
+				arg := raw.(float64)
+				switch op {
+				case ast.ArithmeticOpAdd:
+					result += arg
+				case ast.ArithmeticOpSub:
+					result -= arg
+				case ast.ArithmeticOpMul:
+					result *= arg
+				case ast.ArithmeticOpDiv:
+					result /= arg
+				}
```

Comment:
"You don't get fload mod, because if you need float mod in your terraform config, you're doing something very wrong."

^^ This the logic? Sounds about right to me. :smirk: